### PR TITLE
Improve `separate_circuit` docstring

### PR DESCRIPTION
**Pull request by garrison**
I've also added `circuit_knitting/` to the paths passed to pytest, as this is required for doctests to be discovered.

### CODE REVIEW

File: circuit_knitting/utils/transforms.py [hunk 1]
```diff
@@ -52,15 +64,41 @@ def separate_circuit(
...
 
+    If ``partition_labels`` is provided, then the circuit will be separated
+    according to those labels.  If it is ``None``, then the circuit will be
+    fully separated into its disconnected components, each of which will be
+    labeled with consecutive integers starting with 0.
+
+    >>> qc = QuantumCircuit(4)
+    >>> _ = qc.x(0)
+    >>> _ = qc.cx(1, 2)
+    >>> _ = qc.h(3)
```
Comment (garrison) [inline circuit_knitting/utils/transforms.py:75]: Without the `_ = `, one gets extra output from the non-`None` expressions:

```
>>> from qiskit import QuantumCircuit
>>> qc = QuantumCircuit(4)
>>> qc.x(0)
<qiskit.circuit.instructionset.InstructionSet object at 0x7f710e9921c0>
>>> qc.cx(1, 2)
<qiskit.circuit.instructionset.InstructionSet object at 0x7f70da3873a0>
>>> qc.h(3)
<qiskit.circuit.instructionset.InstructionSet object at 0x7f70da3872b0>
```

and doctest fails unless these are also added here.  Of course, the memory address is not predictable, so there is no way to allow such a thing to be output and still have the tests pass.

File: circuit_knitting/cutting/cutqc/mip_model.py
Comment (garrison) [inline circuit_knitting/cutting/cutqc/mip_model.py:1]: This file had to be modified for doctest discovery to work on Python 3.11 (i.e., when DOcplex is not present).

File: circuit_knitting/utils/transforms.py [hunk 2]
```diff
@@ -40,7 +45,14 @@
 
 
 class SeparatedCircuits(NamedTuple):
-    """Named tuple for result of :class:`separate_circuit`."""
+    """Named tuple for result of :func:`separate_circuit`.
+
+    ``subcircuits`` is a dict of circuits, keyed by each partition label.
+    ``qubit_map`` is a list of length number of qubits in the original circuit.
```
Comment (caleb-johnson) [inline circuit_knitting/utils/transforms.py:51]:
```suggestion
    ``qubit_map`` is a list with length equal to the number of qubits in the original circuit.
```

**Review by caleb-johnson — APPROVED** [review]
Awesome, much needed, thanks. Thanks for adding it to api docs as well

File: circuit_knitting/utils/transforms.py (hunk 2 above)
Comment (garrison) [reply circuit_knitting/utils/transforms.py]: Oh man, thank you.  I rewrote this countless times and was never quite satisfied.

**Pull request merged**